### PR TITLE
CLOUDP-260777: Plugin documentation

### PR DESCRIPTION
**Pull request by stefan4h**
- Added documentation on how to create plugins
- Changed name of binary file
- Removed trigger in release workflow
- rename module to make it easier to use repo as a template

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,69 @@
-# atlas-cli-plugin-example
+# Atlas CLI Plugin Example
+This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 
+
+## How does it work?
+
+A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards
```
Comment (jvincent-mongodb) [inline README.md:6]:
```suggestion
A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards:
```

File: README.md [hunk 2]
```diff
@@ -1 +1,69 @@
...
+This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 
+
+## How does it work?
+
+A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards 
+- all arguments 
+- environment variables
+- standard input/output/error
+
+That allows the plugin to operate as if it were a native part of the Atlas CLI.
```
Comment (jvincent-mongodb) [inline README.md:11]: Suggest cutting this line.

File: README.md [hunk 3]
```diff
@@ -1 +1,69 @@
...
+
+## Developing a Plugin
+
+Plugins can be developed with any programming language, however, in this example plugin we are using Go. A plugin directory can contain any number of files, but it strictly requires two: an executable binary file and a `manifest.yml` file.
```
Comment (jvincent-mongodb) [inline README.md:15]:
```suggestion
Plugins can be developed with any programming language. A plugin directory can contain any number of files, but it strictly requires two: an executable binary file and a `manifest.yml` file.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,69 @@
...
+| `commands.<command_name>.description` | A description of the command | No
+
+> **Important:** 
+> - The `version` field in the manifest file is crucial for the Atlas CLI to identify the installed plugin version and manage updates. Ensure that this field is always aligned with the plugin's release version to enable correct updates.
+> - If your plugin is not part of a Github repository you can omit the entire `github` section. However, if your plugin is hosted on GitHub and you want to enable update functionality, you must include this section.
+
+
+You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create `manifest.yml` file and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.
```
Comment (jvincent-mongodb) [inline README.md:49]:
```suggestion
You may notice that there is no `manifest.yml` file in this repository, only a `manifest.template.yml`. This is because we use a [GitHub Action](https://github.com/mongodb/atlas-cli-plugin-example/blob/master/.github/workflows/release.yml) to automatically generate the `manifest.yml` file. Every time a new release is published, the template is used to create the required `manifest.yml` file, and the release's version is automatically inserted into the `version` field. The [GoReleaser](https://goreleaser.com/) is used to create release assets that contain the plugin's binary and the `manifest.yml`.
```

**Review by jvincent-mongodb — APPROVED** [review]
LGTM with a few suggested phrasing updates.

File: README.md (hunk 2 above)
Comment (stefan4h) [reply README.md:11]: Done

File: README.md [hunk 5]
```diff
@@ -1 +1,66 @@
-# atlas-cli-plugin-example
+# Atlas CLI Plugin Example
+This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 
+
+## How does it work?
+
+A plugin for the Atlas CLI is essentially a standalone CLI. When a plugin is installed, the Atlas CLI downloads a release asset from the plugin's GitHub repository and unpacks it into a plugin directory within the Atlas CLI's config folder. The plugin is then integrated as a standard subcommand, seamlessly extending the core functionality of the CLI. When a plugin command is invoked, the Atlas CLI executes the plugin's binary and forwards:
```
Comment (jeroenvervaeke) [inline README.md:6]: I find this a little hard to follow, what about something among the lines of:

```markdown
### Plugin Execution
A plugin for the Atlas CLI consists of a manifest file and a binary.
The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands is called.

When a subcommand is invoked, the Atlas CLI executes the plugin's binary and forwards:
- all arguments 
- environment variables
- standard input/output/error

### Plugin Installation
1. When a plugin is installed, the Atlas CLI takes the following steps:
download the release asset from the current OS from the plugin GitHub repository
2. unpack the assets (manifest+binary+optinal extras) into the Atlas CLI plugin directory
```

**Review by jeroenvervaeke — CHANGES_REQUESTED** [review]

**Review by jeroenvervaeke — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,75 @@
-# atlas-cli-plugin-example
+# Atlas CLI Plugin Example
+This repository was created for developers who want to extend the [Atlas CLI](https://github.com/mongodb/mongodb-atlas-cli) with their own functionality. 
+
+## How does it work?
+
+### Plugin Execution
+A plugin for the Atlas CLI consists of a manifest file and a binary.
+The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands is called.
```
Comment (jvincent-mongodb) [inline README.md:8]:
```suggestion
The manifest defines which subcommands the plugin exposes, and the binary is called when one of these subcommands are called.
```

File: README.md [hunk 7]
```diff
@@ -1 +1,75 @@
...
+
+When a subcommand is invoked, the Atlas CLI executes the plugin's binary and forwards:
+- all arguments 
+- environment variables
+- standard input/output/error
+
+### Plugin Installation
+When a plugin is installed, the Atlas CLI takes the following steps:
```
Comment (jvincent-mongodb) [inline README.md:16]:
```suggestion
When you install a plugin, the Atlas CLI takes the following steps:
```

File: README.md [hunk 8]
```diff
@@ -1 +1,75 @@
...
+1. download the release asset for the users OS from the plugin GitHub repository
```
Comment (jvincent-mongodb) [inline README.md:17]:
```suggestion
1. download the release asset for the user's OS from the plugin GitHub repository
```
